nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -656,9 +655,15 @@ in
       preStart =
         ''
         ${cfg.preStart}
-        ${cfg.package}/bin/nginx -c ${configFile}/nginx.conf -p ${cfg.stateDir} -t
+        test -d /tmp/.spool/nginx/logs || mkdir -p -m 750 /tmp/.spool/nginx/logs
+        ${cfg.package}/bin/nginx -c ${configFile}/nginx.conf -p /tmp/.spool/nginx -t
         '';
       serviceConfig = {
+        User = cfg.user;
+        Group = cfg.group;
+        AmbientCapabilities = "CAP_NET_BIND_SERVICE";
+        CapabilityBoundingSet = "CAP_NET_BIND_SERVICE";
+        PermissionsStartOnly = true;
```

Comment:
We probably should not introduce new instances of this: https://github.com/NixOS/nixpkgs/issues/53852

Reply:
Thanks. Updated config.